### PR TITLE
[KFSPTS-5301] Make CuYearEndSalaryExpenseTransferDocument processing …

### PR DESCRIPTION
**Pull request by nkimble**
…consistent with standard labor SalaryExpenseTransferDocument processing. Consolidate duplicate year end salary expense transfer spring definition files into a single definition.

### CODE REVIEW

**Comment by kyleoliveira** [comment]
retest this please
^ Hopefully this will retrigger the tests...

**Comment by kyleoliveira** [comment]
retest this please
^ Hopefully this will retrigger the tests...

**Comment by kyleoliveira** [comment]
retest this please

**Comment by kyleoliveira** [comment]
retest this please

**Comment by kyleoliveira** [comment]
retest this please

**Comment by kyleoliveira** [comment]
retest this please

File: src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml [hunk 1]
```diff
@@ -18,20 +18,48 @@
    - along with this program.  If not, see <http://www.gnu.org/licenses/>.
  -->
 
-  <bean id="YearEndSalaryExpenseTransferDocument" parent="YearEndSalaryExpenseTransferDocument-parentBean">
+  <bean id="CuYearEndSalaryExpenseTransferDocument" parent="CuYearEndSalaryExpenseTransferDocument-parentBean"/>
+  <bean id="CuYearEndSalaryExpenseTransferDocument-parentBean" abstract="true" parent="TransactionalDocumentEntry">
```
Comment (jhulslander) [inline src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml]: Should the parent of CuYearEndSalaryExpenseTransferDocument-parentBean be YearEndSalaryExpenseTransferDocument-parentBean?  It looks like it was prior to this change, so we can keep any  inheritance

**Comment by jhulslander** [comment]
This looks good.  One possible change to consider

File: src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml (hunk 1 above)
Comment (nkimble) [inline src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml]: Good catch.

**Comment by nkimble** [comment]
Suggested change made. Please review again when you get a chance.

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
Looks good, after the tests pass, I will merge

File: src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml [hunk 2]
```diff
@@ -18,20 +18,47 @@
    - along with this program.  If not, see <http://www.gnu.org/licenses/>.
  -->
 
-  <bean id="YearEndSalaryExpenseTransferDocument" parent="YearEndSalaryExpenseTransferDocument-parentBean">
+  <bean id="CuYearEndSalaryExpenseTransferDocument" parent="YearEndSalaryExpenseTransferDocument-parentBean">
+    <property name="allowsErrorCorrection" value="true"/>
+    <property name="documentTypeName" value="YEST"/>
     <property name="documentClass" value="edu.cornell.kfs.module.ld.document.CuYearEndSalaryExpenseTransferDocument"/>
+    <property name="baseDocumentClass" value="edu.cornell.kfs.module.ld.document.CuYearEndSalaryExpenseTransferDocument"/>
+    <property name="businessRulesClass" value="org.kuali.kfs.module.ld.document.validation.impl.LaborDocumentRuleEngineRuleBase"/>
```
Comment (bhutchinson) [inline src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml]: Building on Jay's suggestion, unless we override the properties inherited from the parent bean, they don't need to be included. That would leverage the inheritance you now have here, make this file a little cleaner and help us avoid issues if the properties change in base code.

Comment (nkimble) [inline src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml]: Took your suggestion. File now only contains properties to be overridden.

Comment (bhutchinson) [inline src/main/resources/edu/cornell/kfs/module/ld/document/datadictionary/CuYearEndSalaryExpenseTransferDocument.xml]: Great, thanks Nancy!

**Comment by nkimble** [comment]
retest this please

**Pull request merged**